### PR TITLE
Ignore TZID when parsing time with UTC indicator

### LINKED ISSUE #11248: Handle redundant time zone info UTC and timestamp with Z at the end
Calendar subscription with conflicting TZIDS throws:
```
F1TV Live Broacasts - Failed to parse time from 20260725T121500Z, due to conflicting time zone representation. Event has a TZID Etc/GMT and UTC time.
```

In this case we should probably ignore the TZID.

### Test notes
- [x] Import an ics file with an event containing DTSTART with UTC as its TZID and a value with the UTC indication ("Z" at the end of the value), and verify that the event is imported with no errors displayed.

### PR DESCRIPTION
**Pull request by lakshay-2411**
Fixes #11248

`parseTime threw a ParserError when a date-time value had both a TZID parameter and a Z (UTC) suffix, which broke entire calendar subscriptions for non-compliant feeds like F1TV (DTSTART;TZID=Etc/GMT:20260725T121500Z).`

`Per RFC 5545 the Z form is authoritatively UTC, so the TZID is now ignored (with a console warning), matching the zone precedence already used in parseRecurrenceId. The edge-case test asserting the throw was updated to assert the new behavior; genuinely invalid timezones still throw (#11022 behavior preserved).`

### CODE REVIEW

**Comment by ganthern** [comment]
Thank you!

File: src/applications/calendar-app/calendar/export/CalendarParser.ts [hunk 1]
```diff
@@ -959,10 +959,10 @@ export function parseTime(
 	const isAllDay = !("minute" in components)
 
 	if (!isAllDay && components.zone && eventTzid) {
-		throw new ParserError(`Failed to parse time from ${value}, due to conflicting time zone representation. Event has a TZID ${eventTzid} and UTC time.`)
+		console.warn(TAG + ` Time ${value} has both a TZID (${eventTzid}) and a UTC indicator. TZID will be ignored.`)
```
Comment (andrehgdias) [inline src/applications/calendar-app/calendar/export/CalendarParser.ts:937]: We still wanna throw for every other case. Only when eventTzid is UTC we can ignore and not even log.

File: test/tests/calendar/ics-import/CalendarParserTest.ts [hunk 2]
```diff
@@ -274,8 +274,11 @@ o.spec("CalendarParser", function () {
 			})
 		})
 
-		o.test("Edge-case, RFC non-compliant, time with UTC indicator and timezone throws ParserError", function () {
-			o.check(() => parseTime("20260617T214000Z", zone)).throws(ParserError)
+		o.test("Edge-case, RFC non-compliant, time with UTC indicator and timezone ignores the TZID", function () {
```
Comment (andrehgdias) [inline test/tests/calendar/ics-import/CalendarParserTest.ts:278]: So we would need two test cases asserting throws when there is a TZID and the Z at the end.
and one for not throws and proceed to parse ignoring the TZID

**Review by andrehgdias — CHANGES_REQUESTED** [review]
Thanks for being proactive about it!

**Comment by dj-sf** [comment]
Implemented similar behavior while still disallowing TZIDs other than UTC:

https://github.com/tutao/tutanota/commit/fcb7b1fdddb3fa9d6514b0e1de5b7447f6d6197f
